samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java

```diff
@@ -793,12 +796,16 @@ private String submitForGrade(boolean isFromTimer, boolean submitFromTimeoutPopu
  	  List eventLogDataList = eventService.getEventLogData(adata.getAssessmentGradingId());
 	  if(eventLogDataList != null && eventLogDataList.size() > 0) {
 	 	  EventLogData eventLogData= (EventLogData) eventLogDataList.get(0);
-	 	  eventLogData.setErrorMsg(eventLogMessages.getString("no_error"));
+	 	  if (submitFromTimeoutPopup) {
+	 	    eventLogData.setErrorMsg(eventLogMessages.getString("timer_submit"));
+	 	  } else {
+	 	    eventLogData.setErrorMsg(eventLogMessages.getString("no_error"));
+	 	  }
 	 	  Date endDate = new Date();
 	 	  eventLogData.setEndDate(endDate);
 	 	  if(eventLogData.getStartDate() != null) {
 	 	      double minute= 1000*60;
-	 	      int eclipseTime = (int)Math.ceil(((endDate.getTime() - eventLogData.getStartDate().getTime())/minute));
+	 	      int eclipseTime = (int)Math.round(((endDate.getTime() - eventLogData.getStartDate().getTime())/minute));
```

Comment:
changed ceil to round as a 2 minute timer could result in this elapsed time being 2.01 mins which ceilings up to 3